Feature agent calendar

peterhsieh opened the pull request:
- AgentCalendar (View, Dialog, Delete)
- Events Resource API/Endpoints

peterhsieh committed:
Added AgentCalendar Component

- Replaced Stub with AgentCalendar
- AgentCalendar - List Calendar with Toolbar
- CalendarListDay  - Sort Events into Days and display a list
- CalendarListItem - Items in a CalendarListDay
- Necessary API calls/endpoints created
- Ability to view (dialog) and delete

peterhsieh committed:
Code cleanup, empty month note, styling

peterhsieh committed:
Disable edit button (not implemented), styling changes

peterhsieh commented:
## Issues
- Couldn't work out how to sticky the toolbar (attempted to use react.sticky.. but it failed)
- Not sure how to have the list load 'today' at a scrolled position (is that even possible if there aren't enough list items?)

## ToDo
- [ ] Implement the Edit button with Dialog
- [ ] Events in the past are faded (Usability)

nirrek committed:
fix compilation error due to clobbering existing getEvents endpoint

nirrek committed:
Add bottom padding so that final row dropdown fits on screen

nirrek committed:
Update Api endpoint to getAllEvents()

nirrek commented on scripts/components/CalendarListItem.jsx:
```diff
@@ -0,0 +1,114 @@
+      callback: (err, response) => {
+        if (err) {
+        console.log(err);
+        return reply(err);
```
`reply()` is a Hapi.js serverside function.... Not sure why it is being used here.

nirrek committed:
cleanup syntax errors, undefined functions, and formatting

nirrek commented:
To scroll today into view use something like this:

``` javascript
componentDidMount() {
  React.findDOMNode(this.refs.today).scrollIntoView();
}
```

You will have to make sure you place a ref="today" onto the green header bit of today.

As for scrolling to it if there is no scroll position: no there is no simple way to implement this on the web afaik. Would require you being able to scroll past the actual height of the content. So it would be doable, I just don't think you can do it easily with existing APIs.

nirrek committed:
(CalendarListItem) add missing keys

nirrek committed:
(CalendarListDay) add missing keys, cleanup unused code and syntax

nirrek committed:
(AgentCalendar) fix equality check

nirrek commented:
## Todo Before Merging:
- [x] Implement `scrollIntoView` for the current date.
- [x] Fade out past events
- [x] Add ability to edit an event
- [x] Add a snackbar `Event deleted` / `Event updated` notification when performing actions.

nirrek committed:
(CalendarListItem) remove inset on list item divider

peterhsieh committed:
Merge branch 'master' into feature-agent-calendar

peterhsieh committed:
New datepicker, restructure, snackbar support, edit + delete function

- Added React Kronos (datetime picker)
- Rearrange component structure
- Added Snackbar Support
- Added Calendar Event Form Dialog
- Restructure: CalendarListDay now represents a single day of events.
(Moved mapping to AgentCalendar)
- Updated Event API to have update calls

peterhsieh committed:
Merge branch 'master' into feature-agent-calendar

Conflicts:
	assets/agent.png
	assets/marquee.png
	assets/owner.png
	assets/tenant.png
	scripts/app.js

- Conflicts fixed

peterhsieh commented:
**A file in the Kronos causes the building of Bundle.js to fail** - I think it may be crashing with webpack. 
Workaround is to remove all content within the file (react-kronos>nmodules>color> nmodules>color-string>nmodules>color-name>index.json

nirrek committed:
Merge branch 'master' into feature-agent-calendar

nirrek committed:
(CalendarEditEventForm) fix typos breaking the build

nirrek committed:
(AgentCalendar) fix scrolling to today, toolbar width on resize

nirrek committed:
(CalendarEditEventForm) fix kronos, fix stale state bugs, fix dialog to use proper dialog component

nirrek committed:
(CalendarListItem) update to using new CalendarEditEventForm interface

nirrek committed:
(npm) add raf dependency for request animation frame

nirrek committed:
Add utility for smooth scrolling the scroll position of an element

nirrek committed:
(AgentCalendar) improve scrollToToday behaviour
